plugin/lib/class-php-compatibility-checker.php

```diff
@@ -102,73 +98,144 @@ public function init() {
-				// Get array of active plugins.
-				$active_plugins = get_option( 'active_plugins' );
+		if ( empty( $plugins ) ) {
+			return array();
+		}
 
-				if ( ! in_array( $k, $active_plugins, true ) ) {
-					continue;
-				}
-			}
+		$active_plugins = get_option( 'active_plugins' );
+
+		// Add "active" attribute.
+		$plugins = array_map(
+			function( $plugin_file, $plugin_data ) use ( $active_plugins ) {
+				$plugin_data['plugin_file'] = $plugin_file;
+				$plugin_data['active']      = in_array( $plugin_file, $active_plugins, true ) ? 'yes' : 'no';
+				return $plugin_data;
+			},
+			array_keys( $plugins ),
+			$plugins
+		);
 
-			$plugin_file = plugin_dir_path( $k );
+		$plugin_info = get_site_transient( 'update_plugins' );
 
-			// Plugin in root directory (like Hello Dolly).
-			if ( './' === $plugin_file ) {
-				$plugin_path = $plugin_base . $k;
+		// Extract real plugin slugs from the update_plugins transient.
+		foreach ( $plugins as $key => $plugin_data ) {
+			$plugin_file = $plugin_data['plugin_file'];
+
+			if ( isset( $plugin_info->response[ $plugin_file ] ) ) {
+				$plugins[ $key ]['slug'] = $plugin_info->response[ $plugin_file ]->slug;
+			} elseif ( isset( $plugin_info->no_update[ $plugin_file ] ) ) {
+				$plugins[ $key ]['slug'] = $plugin_info->no_update[ $plugin_file ]->slug;
 			} else {
-				$plugin_path = $plugin_base . $plugin_file;
-			}
-			$id[] = array( $plugin_file );
-
-			if ( is_object( $update_plugins ) && is_array( $update_plugins->response ) ) {
-				// Check for plugin updates.
-				foreach ( $update_plugins->response as $uk => $uv ) {
-					// If we have a match.
-					if ( $uk === $k ) {
-						$this->debug_log( 'An update exists for: ' . $v['Name'] );
-						// Save the update version.
-						//update_post_meta( $id, 'update', $uv->new_version );
-						// Save the current version.
-						//update_post_meta( $id, 'version', $v['Version'] );
-					}
-				}
+				$plugins[ $key ]['slug'] = false;
 			}
-		}//end foreach
-		return $id;
+		}
+
+		// Compact output.
+		$plugins = array_map(
+			function( $plugin ) {
+				return array(
+					'slug'    => sanitize_text_field( $plugin['slug'] ),
+					'name'    => sanitize_text_field( $plugin['Name'] ),
+					'version' => sanitize_text_field( $plugin['Version'] ),
+					'active'  => $plugin['active'],
+				);
+			},
+			$plugins
+		);
+
+		/**
+		 * Filter which plugins should be scanned.
+		 *
+		 * @param array[] $plugins {
+		 *     List of plugins.
+		 *
+		 *     @type string $slug Plugin slug.
+		 *     @type string $name Plugin name.
+		 *     @type string $version Plugin version.
+		 *     @type string $active Whether the plugin is active (yes or no).
+		 * }
+		 */
+		return apply_filters( 'phpcompat_plugins_to_scan', $plugins );
+	}
+
+	/**
+	 * Get all themes that should be scanned.
+	 *
+	 * @return array
+	 */
+	public function get_themes_to_scan() {
+		$themes_data = wp_prepare_themes_for_js();
+
+		$themes = array_map(
```

Comment:
For feature parity sake, what do you think about adding similar exclusion ability here for themes? Something like:
```suggestion
                /**
		 * Filter which themes should be excluded from scans.
		 *
		 * This will exclude based on the theme name, not the theme slug.
		 *
		 * @param string[] $excluded_themes Themes we want to exclude.
		 */
		$excluded_themes = apply_filters( 'phpcompat_excluded_themes', array() );

		// Exclude some themes.
		$themes_data = array_filter(
			$themes_data,
			function ( $theme_data ) use ( $excluded_themes ) {
				return ! in_array( $theme_data['name'], $excluded_themes, true );
			}
		);

		$themes = array_map(
```